### PR TITLE
Fix issue when converting datetimes and pandas Timestamps in an expression

### PR DESCRIPTION
**Pull request by cpcloud**
cc @mrocklin

### CODE REVIEW

**Comment by cpcloud** [comment]
@mrocklin okay with this?

File: blaze/compute/pyfunc.py [hunk 1]
```diff
@@ -63,7 +64,7 @@ def _print_python(expr, leaves=None):
 
 @dispatch((datetime.datetime, datetime.date))
 def _print_python(expr, leaves=None):
-    return repr(expr), {'datetime': datetime}
+    return repr(expr), {'datetime': datetime, 'Timestamp': pd.Timestamp}
```
Comment (mrocklin) [inline blaze/compute/pyfunc.py]: Are there cases where we intended to print python datetime objects that we actually end up printing Pandas Timestamps?  If so is there an issue?  Do we ever not assume that pandas is around?

File: blaze/expr/arithmetic.py [hunk 2]
```diff
@@ -220,7 +221,7 @@ def scalar_coerce(_, val):
 
 @dispatch(ct.DateTime, _strtypes)
 def scalar_coerce(_, val):
-    return dt_parse(val)
+    return pd.Timestamp(val)
```
Comment (mrocklin) [inline blaze/expr/arithmetic.py]: Does this work nicely with sqlalchemy?

Comment (cpcloud) [inline blaze/expr/arithmetic.py]: In what sense?

```python
In [26]: import sqlalchemy as sa

In [27]: c = sa.Column('col', sa.types.TIMESTAMP)

In [28]: expr = c >= pd.Timestamp('now').to_pydatetime()

In [29]: print(expr.compile().render_literal_bindparam(expr.right))
```
raises `NotImplementedError`, and so does the version with the `Timestamp` instance.

Comment (mrocklin) [inline blaze/expr/arithmetic.py]: Hrm, that's unfortunate.  I'm surprised that sqlalchemy doesn't manage python datetimes well.

Comment (cpcloud) [inline blaze/expr/arithmetic.py]: It handles them just fine, except when rendering literal value. You can use `literal_column(dt.strftime('...'))` or it's up to the dialect to manage literal rendering. You can of course pass `datetime` around in expressions and it deals with sending those values to the database quite nicely.

File: blaze/compute/pyfunc.py (hunk 1 above)
Comment (cpcloud) [inline blaze/compute/pyfunc.py]: > Are there cases where we intended to print python datetime objects that we actually end up printing Pandas Timestamps?

Not sure. FWIW, `Timestamp` is a subclass of `datetime.datetime`.

> Do we ever not assume that pandas is around?

No, pandas is a hard dependency.

File: blaze/expr/arithmetic.py (hunk 2 above)
Comment (cpcloud) [inline blaze/expr/arithmetic.py]: As an example, I render literal datetime date and timedelta values in `kdbpy` in the way that kdbpy expects them

**Comment by cpcloud** [comment]
I'll add that the thing that's nice about pandas `Timestamp`s is that they can handle the precision of other systems' timestamps, such as kdb which also has nanosecond precision timestamp types.

**Comment by cpcloud** [comment]
merging

**Pull request merged**